src/memberBundle.spec.ts

```diff
@@ -0,0 +1,140 @@
+
+      const selfIssueOrganisationCertificateResult = new ArrayBuffer(1);
+      selfIssueOrganisationCertificate.mockResolvedValueOnce(selfIssueOrganisationCertificateResult);
+      const issueMemberCertificateResult = new ArrayBuffer(2);
+      issueMemberCertificate.mockResolvedValueOnce(issueMemberCertificateResult);
+      const retrieveVeraDnssecChainResult = new ArrayBuffer(3);
+      retrieveVeraDnssecChain.mockResolvedValueOnce(retrieveVeraDnssecChainResult);
+      const serialiseMemberIdBundleResult = new ArrayBuffer(4);
+      serialiseMemberIdBundle.mockReturnValueOnce(serialiseMemberIdBundleResult);
+
+      await orgModel.create({
+        name: ORG_NAME,
+        privateKeyRef: orgPrivateKeyRef,
+        publicKey: orgPublicKey,
+      });
+
+      const member = await memberModel.create({
+        orgName: ORG_NAME,
+        name: MEMBER_NAME,
+        role: Role.REGULAR,
+      });
+      const memberPublicKey = await memberPublicKeyModel.create({
+        memberId: member.id,
+        publicKey: memberPublicKeyBuffer,
+        serviceOid: TEST_SERVICE_OID,
+      })
+
+      const result = await generateMemberBundle(memberPublicKey._id.toString(), serviceOptions);
+
+      requireSuccessfulResult(result);
+      expect(result.result).not.toBeNull()
+      const plus90Days = addDays(new Date(), 90);
+
+      const selfIssueOrganisationCertificateParameters = selfIssueOrganisationCertificate.mock.calls[0];
+      expect(selfIssueOrganisationCertificateParameters[0]).toBe(ORG_NAME);
+      const calledKeyPair = selfIssueOrganisationCertificateParameters[1];
+      const orgPrivateKeyBuffer = await kmsForKeys.getPrivateKeyRef(calledKeyPair.privateKey);
+      const orgPublicKeyBuffer = await derSerialisePublicKey(calledKeyPair.publicKey);
+      expect(orgPrivateKeyBuffer.toString()).toBe(orgPrivateKeyRef.toString())
+      expect(orgPublicKeyBuffer.toString()).toBe(orgPublicKey.toString())
+      expect(selfIssueOrganisationCertificateParameters[2]).toBeBetween(subSeconds(plus90Days, 20), addSeconds(plus90Days, 20));
+
+
+      const issueMemberCertificateParameters = issueMemberCertificate.mock.calls[0];
+      expect(issueMemberCertificateParameters[0]).toBe(MEMBER_NAME);
+      expect((await derSerialisePublicKey(issueMemberCertificateParameters[1])).toString()).toBe(memberPublicKeyBuffer.toString());
+      expect(issueMemberCertificateParameters[2]).toBe(selfIssueOrganisationCertificateResult);
+      expect(issueMemberCertificateParameters[3]).toBe(calledKeyPair.privateKey);
+      expect(issueMemberCertificateParameters[4].getTime()).toBe(selfIssueOrganisationCertificateParameters[2].getTime());
+
+      expect(retrieveVeraDnssecChain).toHaveBeenCalledOnceWith(ORG_NAME);
+      expect(serialiseMemberIdBundle).toHaveBeenCalledOnceWith(issueMemberCertificateResult, selfIssueOrganisationCertificateResult, retrieveVeraDnssecChainResult);
+      expect(result.result).toBe(serialiseMemberIdBundleResult);
+
+
+      });
```

Comment:
The test looks ugly but this way every method is tested. Maybe we can split the test into multiple to make it more readable